Source/PeleLM.cpp

```diff
@@ -1328,8 +1328,33 @@ PeleLM::init_mixture_fraction()
                for (int i=0; i<NUM_SPECIES; ++i) {
                   XF[i] = compositionIn[i];
                }
-               EOS::X2Y(XO, YO);
-               EOS::X2Y(XF, YF);
+               // Here comes the fun part for calling DEVICE functions from HOST
+               amrex::Gpu::DeviceVector<amrex::Real> XO_v(NUM_SPECIES);
+               amrex::Gpu::DeviceVector<amrex::Real> XF_v(NUM_SPECIES);
+               for (int i = 0; i < NUM_SPECIES; i++ ) {
+                  XO_v[i] = XO[i];
+                  XF_v[i] = XF[i];
+               }
+               amrex::Gpu::DeviceVector<amrex::Real> YO_v(NUM_SPECIES);
+               amrex::Gpu::DeviceVector<amrex::Real> YF_v(NUM_SPECIES);
+               amrex::Real* XO_d = XO_v.data();
+               amrex::Real* XF_d = XF_v.data();
+               amrex::Real* YO_d = YO_v.data();
+               amrex::Real* YF_d = YF_v.data();
+               Box dumbx({AMREX_D_DECL(0,0,0)},{AMREX_D_DECL(0,0,0)});
+               amrex::ParallelFor(dumbx, [XO_d,XF_d,YO_d,YF_d]
+               AMREX_GPU_DEVICE(int i, int j, int k) noexcept
+               {
+                  EOS::X2Y(XO_d,YO_d);
+                  EOS::X2Y(XF_d,YF_d);
+               });
+#if AMREX_USE_GPU
+               amrex::Gpu::dtoh_memcpy(YO,YO_d,sizeof(amrex::Real)*NUM_SPECIES);
+               amrex::Gpu::dtoh_memcpy(YF,YF_d,sizeof(amrex::Real)*NUM_SPECIES);
+#else
+               std::memcpy(YO,YO_d,sizeof(amrex::Real)*NUM_SPECIES);
+               std::memcpy(YF,YF_d,sizeof(amrex::Real)*NUM_SPECIES);
+#endif
```

Comment:
Dang, how awful.

Reply:
Yes, calling AMREX_GPU_DEVICE functions from the host is now pretty awful... The only other way I see is to have a duplicated set of pure host functions, but it's as bad I think.